rotkehlchen/exchanges/bitstamp.py

```diff
@@ -0,0 +1,711 @@
+                        f'Failed to deserialize a Bitstamp {case_pretty}. '
+                        f'Check logs for details. Ignoring it.',
+                    )
+                # else:
+                results.append(result)
+                has_results = True  # NB: endpoint agnostic
+
+            if len(response_list) < limit or is_result_timesamp_gt_end_ts:
+                break
+
+            # NB: update pagination params per endpoint
+            if endpoint == 'user_transactions':
+                # NB: re-assign dict instead of update, prevent lose call args values
+                call_options = call_options.copy()
+                offset = 0 if has_results else call_options['offset'] + API_MAX_LIMIT
+                since_id = int(results[-1].link) + 1 if has_results else call_options['since_id']
+                call_options.update({
+                    'since_id': since_id,
+                    'offset': offset,
+                })
+
+        return results
+
+    def _deserialize_trade(
+            self,
+            raw_trade: Dict[str, Any],
+    ) -> Trade:
+        """Process a trade user transaction from Bitstamp and deserialize it.
+
+        Can raise Can raise DeserializationError.
+        """
+        timestamp = iso8601ts_to_timestamp(raw_trade['datetime'])
+        trade_pair_data = self._get_trade_pair_data_from_transaction(raw_trade)
+        base_asset_amount = deserialize_asset_amount(
+            raw_trade[trade_pair_data.base_asset_symbol],
+        )
+        quote_asset_amount = deserialize_asset_amount(
+            raw_trade[trade_pair_data.quote_asset_symbol],
+        )
+
+        if base_asset_amount >= ZERO:
+            trade_type = TradeType.BUY
+            amount = base_asset_amount
+            fee_currency = trade_pair_data.quote_asset
+        else:
+            trade_type = TradeType.SELL
+            amount = quote_asset_amount
+            fee_currency = trade_pair_data.base_asset
+
+        trade = Trade(
+            timestamp=timestamp,
+            location=Location.BITSTAMP,
+            pair=trade_pair_data.trade_pair,
+            trade_type=trade_type,
+            amount=amount,
+            rate=deserialize_price(raw_trade[trade_pair_data.pair]),
+            fee=deserialize_fee(raw_trade['fee']),
+            fee_currency=fee_currency,
+            link=str(raw_trade['id']),
+            notes='',
+        )
+        return trade
+
+    def _deserialize_asset_movement(
+            self,
+            raw_movement: Dict[str, Any],
+    ) -> AssetMovement:
+        """Process a deposit/withdrawal user transaction from Bitstamp and
+        deserialize it.
+
+        Can raise Can raise DeserializationError.
+
+        From Bitstamp documentation, deposits/withdrawals can have a fee
+        (the amount is expected to be in the currency involved)
+        https://www.bitstamp.net/fee-schedule/
+
+        TODO: WIP, waiting for Binance Support clarification about the JSON
```

Comment:
As I've explained we will need some clarification about the JSON.

Reply:
It seems you got it now? Can you verify that your logic is correct from the response?